lib/service.rb

```diff
@@ -484,6 +484,8 @@ def inherited(svc)
 
   attr_reader :remote_calls
 
+  attr_accessor :needs_public_key_signature, :public_key
```

Comment:
It might be nicer to decouple this a bit from the public key. Maybe a `sign` block could be configured instead. Further up the stack we could say

```ruby
svc.sign = lambda { |payload_string|
  raw_sig = key.sign(message: payload_string, raw: true).signature
  return Base64.strict_encode64(raw_sig)
}
```

and then in this repo we'd just do

```ruby
if needs_public_key_signature
  http.headers['GITHUB-PUBLIC-KEY-SIGNATURE'] = sign(body)
end
```

Reply:
This is an excellent idea, @mastahyeti. I took it a few steps further in e03332d...dddd1ac.